internal/handlers/common/compare.go

```diff
@@ -47,72 +48,92 @@ func compareScalars(a, b any) compareResult {
 	case float64:
 		switch b := b.(type) {
 		case float64:
+			if math.IsNaN(a) && math.IsNaN(b) {
+				return equal
+			}
 			return compareOrdered(a, b)
 		case int32:
 			return compareNumbers(a, int64(b))
 		case int64:
 			return compareNumbers(a, b)
 		default:
-			panic(fmt.Sprintf("unexpected type %T", b))
+			return notEqual
 		}
 
 	case string:
-		b := b.(string)
-		return compareOrdered(a, b)
+		b, ok := b.(string)
+		if ok {
+			return compareOrdered(a, b)
+		}
+		return notEqual
 
 	case types.Binary:
-		b := b.(types.Binary)
-		al, bl := len(a.B), len(b.B)
-		if al != bl {
-			return compareOrdered(al, bl)
-		}
-		if a.Subtype != b.Subtype {
-			return compareOrdered(a.Subtype, b.Subtype)
-		}
-		switch bytes.Compare(a.B, b.B) {
-		case 0:
-			return equal
-		case -1:
-			return less
-		case 1:
-			return greater
-		default:
-			panic("unreachable")
+		b, ok := b.(types.Binary)
+		if ok {
```

Comment:
```suggestion
		if !ok {
		  return notEqual
		}
```

Same idea as in https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow – unindent logic, return early

The same in other places

Reply:
Suggestions have been applied.